egs/tedlium/s5_r2/local/ted_train_lm.sh

```diff
@@ -36,6 +37,18 @@ export PATH=$KALDI_ROOT/tools/pocolm/scripts:$PATH
 
 num_dev_sentences=10000
 
+bypass_metaparam_optim_opt=
+# If you want to bypass the metaparameter optimization steps with specific metaparameters
+# un-comment the following line, and change the numbers to some appropriate values.
+# You can find the values from output log of train_lm.py.
+# These example numbers of metaparameters is for 4-gram model (with min-counts)
+# running with train_lm.py.
+# The dev perplexity should be close to the non-bypassed model.
+#bypass_metaparam_optim_opt="--bypass-metaparameter-optimization=0.794,0.022,0.761,0.083,0.043,0.013,1.000,0.359,0.123,0.078,1.000,0.613,0.227,0.227"
```

Comment:
>
> Ok, I'll try to do that. But I also found that the bypass metaparameters
> are different each time even with the same setting. Is this normal?
>

I wouldn't have expected them to be different- I'm not sure.  But it may
not be fatal.

> And after I got those parameters after one time run, I should delete
> existing models and rerun the script with those parameters and expect
> getting similar ppls right?
>

Yes.
Dan



> Ke On Mon, Aug 15, 2016 at 3:59 PM, Daniel Povey <notifications@github.com>
> wrote:
> … <#m_216258476580869229_>
> In egs/tedlium/s5_r2/local/ted_train_lm.sh <#964 (comment)
> <https://github.com/kaldi-asr/kaldi/pull/964#discussion_r74823890>>: > @@
> -36,6 +37,18 @@ export PATH=$KALDI_ROOT/tools/pocolm/scripts:$PATH > >
> num_dev_sentences=10000 > > +bypass_metaparam_optim_opt= > +# If you want
> to bypass the metaparameter optimization steps with specific metaparameters
> > +# un-comment the following line, and change the numbers to some
> appropriate values. > +# You can find the values from output log of
> train_lm.py. > +# These example numbers of metaparameters is for 4-gram
> model (with min-counts) > +# running with train_lm.py. > +# The dev
> perplexity should be close to the non-bypassed model. >
> +#bypass_metaparam_optim_opt="--bypass-metaparameter-
> optimization=0.794,0.022,0.761,0.083,0.043,0.013,1.000,0.
> 359,0.123,0.078,1.000,0.613,0.227,0.227" It must be due to rounding. You
> should modify the code that prints out the 'bypass' metaparameters so that
> if any parameter would be printed as 1.000 it gets printed as 1.999, and
> 0.0 goes to 0.001. All parameters are supposed to be in the range (0, 1).
> Dan … <#m_3557986115878049735_> On Sun, Aug 14, 2016 at 11:15 PM, Ke Li
> ***@***.***> wrote: In egs/tedlium/s5_r2/local/ted_train_lm.sh <#964
> <https://github.com/kaldi-asr/kaldi/pull/964> (comment) <#964 (comment)
> <https://github.com/kaldi-asr/kaldi/pull/964#discussion_r74723538>>>: >
> @@ -36,6 +37,18 @@ export PATH=$KALDI_ROOT/tools/pocolm/scripts:$PATH > >
> num_dev_sentences=10000 > > +bypass_metaparam_optim_opt= > +# If you want
> to bypass the metaparameter optimization steps with specific metaparameters
> > +# un-comment the following line, and change the numbers to some
> appropriate values. > +# You can find the values from output log of
> train_lm.py. > +# These example numbers of metaparameters is for 4-gram
> model (with min-counts) > +# running with train_lm.py. > +# The dev
> perplexity should be close to the non-bypassed model. >
> +#bypass_metaparam_optim_opt="--bypass-metaparameter-
> optimization=0.794,0.022,0.761,0.083,0.043,0.013,1.000,0.
> 359,0.123,0.078,1.000,0.613,0.227,0.227" I just noticed that there is
> 1.000 (which should be invalid) in the bypass metaparameters. I will double
> check that before setting them as default. — You are receiving this because
> you commented. Reply to this email directly, view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/964/files/ f37d8f7
> <https://github.com/kaldi-asr/kaldi/commit/f37d8f73b65a7ab16a3d862e0f91175efe5ac79a>#r74723538>,
> or mute the thread < https://github.com/notifications/unsubscribe-auth/
> ADJVu3MPaSi0C2NxahKRCsuqzZdkQdo6ks5qgAPkgaJpZM4JkFeX> . — You are
> receiving this because you authored the thread. Reply to this email
> directly, view it on GitHub <https://github.com/kaldi-asr/
> kaldi/pull/964/files/f37d8f73b65a7ab16a3d862e0f91175efe5ac79a#r74823890>,
> or mute the thread <https://github.com/notifications/unsubscribe-auth/
> ANVxSoxOqlqdgDipYIldzykbOaox03Uxks5qgMU3gaJpZM4JkFeX> .
> -- Ke Li Dept. of Electrical and Computer Engineering Johns Hopkins
> University Email: kli26@jhu.edu
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/964/files/f37d8f73b65a7ab16a3d862e0f91175efe5ac79a#r74870275>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu4ntVTjMntUUjcaj44vcFU6XwcSUks5qgSl5gaJpZM4JkFeX>
> .
>